services/dynamic/dynamic-xml.service.js

```diff
@@ -47,14 +36,48 @@ module.exports = class DynamicXml extends BaseService {
     } catch (e) {
       throw new InvalidParameter({ prettyMessage: e.message })
     }
-    values = values.map((node, i) =>
-      pathIsAttr ? node.value : node.firstChild.data
-    )
+
+    if (!Array.isArray(values)) {
+      throw new InvalidResponse({
+        prettyMessage: 'must be a collection of elements',
+      })
+    }
+
+    values = values.reduce((accum, node) => {
+      if (pathIsAttr) {
+        accum.push(node.value)
+      } else if (node.firstChild) {
+        accum.push(node.firstChild.data)
+      }
+
+      return accum
+    }, [])
 
     if (!values.length) {
       throw new InvalidResponse({ prettyMessage: 'no result' })
     }
 
-    return renderDynamicBadge({ value: values, prefix, suffix })
+    return { values }
+  }
+
+  async handle(namedParams, { url, query: pathExpression, prefix, suffix }) {
+    // e.g. //book[2]/@id
+    const pathIsAttr = (
```

Comment:
What do you think about moving this to `transform` function? `pathIsAttr` it only used inside that function.

Reply:
Query `//book[1]/title` for an empty element ` <title></title>` also results with `Cannot read property 'data' of null`.